### PR TITLE
ci: add deploy docs

### PR DESCRIPTION
**Pull request by jangsc0000**
Deploy release documentation automatically.

### CODE REVIEW

File: .github/workflows/deploy_docs.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        meson compile -C _build
+
+    - name: Build echo
+      run:  echo ${{ github.ref }}
+
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./_build/doc/html-out
```
Comment (XinfengZhang) [inline .github/workflows/deploy_docs.yml:44]: now sure about this step, could update the https://github.com/intel/libva/tree/gh-pages directly?
or just simulate it? I dont want to update the doc directly, 
will update it with releases.

File: .github/workflows/deploy_docs.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+
+name: deploy docs
+on:
+  release:
+    types: [published]
```
Comment (jangsc0000) [inline .github/workflows/deploy_docs.yml:5]: name: deploy docs
on:
  release:
    types: [published]

File: .github/workflows/deploy_docs.yml (hunk 1 above)
Comment (jangsc0000) [reply .github/workflows/deploy_docs.yml:44]: When you release, the documentation in the release commit is updated.

File: .github/workflows/deploy_docs.yml (hunk 2 above)
Comment (jangsc0000) [inline .github/workflows/deploy_docs.yml:5]: This workflow only works when released by this part.

File: .github/workflows/deploy_docs.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          libx11-xcb-dev \
+           libxcb-dri3-0 \
+          doxygen 
+    - uses: actions/setup-python@v1
+      with:
+        python-version: '3.10' 
+
+    - name: Build
+      run: |
+        python -m pip install ninja
```
Comment (dvrogozh) [inline .github/workflows/deploy_docs.yml:32]: why install via pip vs. from ubuntu packages?

File: .github/workflows/deploy_docs.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        sudo apt-get install -y \
+          libdrm-dev \
+          libegl1-mesa-dev \
+          libgl1-mesa-dev \
+          libx11-dev \
+          libxext-dev \
+          libxfixes-dev \
+          libwayland-dev \
+          libx11-xcb-dev \
+           libxcb-dri3-0 \
```
Comment (dvrogozh) [inline .github/workflows/deploy_docs.yml:24]: fix alignment, please

File: .github/workflows/deploy_docs.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        python -m pip install ninja
+        python -m pip install meson
+        meson _build -Denable_docs=true
+        meson compile -C _build
+
+    - name: Build echo
+      run:  echo ${{ github.ref }}
+
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v3
```
Comment (dvrogozh) [inline .github/workflows/deploy_docs.yml:41]: I did not look closely into github pages, but there seems to be standard ways of doing that. Why use custom github action?

* https://github.com/actions/deploy-pages
* https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site
* https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site

File: .github/workflows/deploy_docs.yml (hunk 2 above)
Comment (dvrogozh) [reply .github/workflows/deploy_docs.yml:5]: what will happen if we release 2.17.0, ..., 2.17.4, 2.18.0 and then release 2.17.5? will 2.17.5 override 2.18.0 posting?fff

**Review by dvrogozh — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy_docs.yml (hunk 5 above)
Comment (jangsc0000) [reply .github/workflows/deploy_docs.yml:40]: I tried this, but I can't deploy with release tag.

File: .github/workflows/deploy_docs.yml [hunk 6]
```diff
@@ -1,15 +1,19 @@
...
+  build:
     runs-on: ubuntu-latest
-    steps:
+    steps: 
     - uses: actions/checkout@v3
-    - name: install prerequisites
+    - id: tag_name
+      name: Get latest tag_name
+      run: |        
+        echo "GIT_LATEST_TAG=`echo $(git tag | grep '^[0-9]\+\.[0-9]\+\?\.[0-9]\+\?$' | sort -V -r | head -n1)`" >> $GITHUB_ENV
```
Comment (jangsc0000) [inline .github/workflows/deploy_docs.yml:15]: By checking the tag, only the latest version of the document is updated.

File: .github/workflows/deploy_docs.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: install prerequisites
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y \
```
Comment (evelikov) [inline .github/workflows/deploy_docs.yml:19]: A better option IMHO is to use:
 - apt-get build-dep libva
 - apt-get install extra-deps-not-installed-with-above

File: .github/workflows/deploy_docs.yml (hunk 3 above)
Comment (evelikov) [reply .github/workflows/deploy_docs.yml]: Ninja is not a python module so the above does not make sense. Ideally both ninja and meson (below) should be installed with "apt-get install ..." if needed

File: .github/workflows/deploy_docs.yml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+    - name: Build
+      run: |
+        python -m pip install ninja
+        python -m pip install meson
+        meson _build -Denable_docs=true
+        meson compile -C _build
+
+    - name: Build echo
+      run:  echo ${{ github.ref }}
```
Comment (evelikov) [inline .github/workflows/deploy_docs.yml]: This is a no-op - drop?

Comment (jangsc0000) [reply .github/workflows/deploy_docs.yml]: It was removed

Comment (evelikov) [reply .github/workflows/deploy_docs.yml]: The change should be squashed appropriately

**Pull request closed without merging**